File: sylphrena-core/syl_core.cpp
/*!
 * Sylphrena AI core program - https://github.com/ShardAi
 * Version - 1.0.0.0
 *
 * Copyright (c) 2017 Eirik Skjeggestad Dale
 */

#include "syl_core.h"

sylCore::sylCore()
{
	counter = 0;
}

sylCore::~sylCore()
{
    delete server;
}

void sylCore::startServer()
{
	server = new sylSocketServer(this);
}

void sylCore::counterMethod()
{
	counter++;
}

void sylCore::msgReceived(string msg)
{
    //TODO: Use msg!
    if(msg.length() < 13)
        return;
    cout << "EIRIKDEBUG PRINTING. " << endl;
    vector<string> tokens = sylLib::split(msg, ' ');

    for(vector<string>::iterator it = tokens.begin(); it != tokens.end(); ++it)
    {
        //TODO run command to neural network IF it is valid.
        cout << "{cmd: " << *it << ", cmd.length(): " << (*it).length() << ".} ";
    }
    cout << endl;
}

File: sylphrena-core/src/core/syl_core.rs
/*!
 * Sylphrena AI core program - https://github.com/ShardAi
 * Version - 1.0.0.0
 *
 * Copyright (c) 2017 Eirik Skjeggestad Dale
 */

use core::syl_tcp::Syltcp;

const LOCAL: &str = "127.0.0.1:8888";

pub struct Sylcore {
    id: String,
    tcp_server: Syltcp,
}

impl Sylcore {
    pub fn new(id: &str) -> Sylcore {
        let core = Sylcore{
            id: id.to_string(),
            tcp_server: Syltcp::new("syl-tcp", LOCAL)
        };
        println!("Created {0}!", core.id.to_string());
        return core;
    }
    pub fn start_server(&mut self) {
        self.tcp_server.start_server();
    }
}